src/components/Messages/PreviewList.tsx

```diff
@@ -44,12 +45,18 @@ const PreviewList: FC<Props> = ({ className, selectedConversationKey }) => {
   const showAuthenticating = currentProfile && authenticating;
   const showLoading = loading && (messages.size === 0 || profiles.size === 0);
 
+  let hasMessages = false;
   const sortedProfiles = Array.from(profiles).sort(([keyA], [keyB]) => {
-    const messageA = messages.get(keyA);
-    const messageB = messages.get(keyB);
-    return (messageA?.sent?.getTime() || 0) >= (messageB?.sent?.getTime() || 0) ? -1 : 1;
+    const aTime = messages.get(keyA)?.sent.getTime() || 0;
+    const bTime = messages.get(keyB)?.sent.getTime() || 0;
+    hasMessages = hasMessages || aTime > 0 || bTime > 0;
+    return aTime >= bTime ? -1 : 1;
   });
 
+  if (hasMessages) {
+    clearMessagesBadge(currentProfile.id);
```

Comment:
We should wrap this in a useEffect. Having things that mutate the state inside the render function can lead to some funky behaviour like infinite render loops.